src/inmanta/user_setup.py

```diff
@@ -0,0 +1,163 @@
+            click.echo("Add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+
+        raise click.ClickException("Make sure signing configuration is added to the config. See the documentation for details.")
+
+    click.echo(f"{'Authentication signing config: ' : <50}{click.style('enabled', fg='green')}")
+
+    # TODO: verify web-console config (if any)
+
+
+async def get_database_connection() -> asyncpg.pool.Pool:
+    database_host = server_config.db_host.get()
+    database_port = server_config.db_port.get()
+
+    database_username = server_config.db_username.get()
+    database_password = server_config.db_password.get()
+    connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+    connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+    connection_timeout = server_config.db_connection_timeout.get()
+    return await data.connect(
+        database_host,
+        database_port,
+        server_config.db_name.get(),
+        database_username,
+        database_password,
+        False,
+        connection_pool_min_size=connection_pool_min_size,
+        connection_pool_max_size=connection_pool_max_size,
+        connection_timeout=connection_timeout,
+    )
+
+
+async def do_user_setup() -> None:
+    """Perform the user setup that requires the database interaction"""
+    connection = None
+    try:
+        connection = await get_database_connection()
+        DBschema = schema.DBSchema(CORE_SCHEMA_NAME, PACKAGE_WITH_UPDATE_FILES, connection)
+        schema_up_to_date = await DBschema.is_DB_schema_up_to_date()
+        if schema_up_to_date:
+            click.echo(f"{'DB up to date' : <50}{click.style('yes', fg='green')}")
+        else:
+            click.echo(f"{'DB up to date' : <50}{click.style('no: please migrate your DB to the latest version', fg='red')}")
+            return
+        users = await data.User.get_list()
+
+        if len(users):
+            raise click.ClickException(
+                "There are already users in the database. If you want to reset the password, use the --reset option."
+            )
+
+        username = click.prompt("What username do you want to use?", default="admin")
+        password = click.prompt("What password do you want to use?", hide_input=True)
+
+        pw_hash = nacl.pwhash.str(password.encode())
+
+        # insert the user
+        user = data.User(
+            username=username,
+            password_hash=pw_hash.decode(),
+            enabled=True,
+            auth_method="password",
+        )
+        await user.insert()
+
+        click.echo(f"{'User %s: ' %username : <50}{click.style('created', fg='green')}")
+    finally:
+        if connection is not None:
+            await data.disconnect()
+
+    click.echo("Make sure to (re)start the orchestrator to activate all changes.")
+
+
+@click.command(help="Do the initial user setup")
```

Comment:
I think we should clearly mention that this command is intended to be executed on the orchestrator itself and not remotely. Maybe we should also ask this as a question to the user on the setup checks.

Reply:
you mean asking "Are you on the orchestrator?" 
=> yes: continue
=> no: abort
?

Reply:
Essentially yes. But I would provide more context in the question. Something like: `This command should be execute locally on the orchestrator you want to configure. Are you running this command locally?`